### PR TITLE
DL3 energy-dependent alpha cut

### PR DESCRIPTION
**Pull request by SeiyaNozaki**
This PR implements energy-dependent alpha cut for the IRF/DL3 production.
The same solution as energy-dependent g/h cuts is basically used for the energy-dependent alpha cut.
(create a separate temporary header to save energy-dependent alpha cuts in IRF)
Since the alpha cut is already applied to observational DL3 data, there is no need to modify gammapy.

### CODE REVIEW

File: lstchain/tools/lstchain_create_irf_files.py [hunk 1]
```diff
@@ -484,11 +508,17 @@ def start(self):
                         'deg'
                     )
             else:
-                extra_headers["AL_CUT"] = (
-                    self.cuts.global_alpha_cut,
-                    'deg'
-                )
-
+                if self.energy_dependent_alpha:
+                    extra_headers["ALPHA_CONT"] = (
```
Comment (chaimain) [inline lstchain/tools/lstchain_create_irf_files.py:512]: I think there is still the 8 character limit on the HIERARCH card. In the latest astropy there is support for [longer keywords](https://docs.astropy.org/en/stable/io/fits/usage/headers.html#hierarch-cards), but we do not currently have the latest astropy.

File: lstchain/io/event_selection.py [hunk 2]
```diff
@@ -178,6 +178,50 @@ def apply_energy_dependent_theta_cuts(self, data, theta_cuts):
...
+        """
+        Applying a given energy-dependent alpha cuts to a data file, along the
+        reco energy bins provided.
+        """
+        
+        data["selected_alpha"] = evaluate_binned_cut(
+            data["alpha"],
+            data["reco_energy"],
+            alpha_cuts,
+            operator.le,
```
Comment (chaimain) [inline lstchain/io/event_selection.py:221]: The operation should be consistent with the operation we use when applying global alpha cuts. I have made a commit to this fix in PR #943. There is no need to apply the minor fix here.

**Review by chaimain — COMMENTED** [review]
Sorry for the late review.
It looks good to me in general.

File: lstchain/tools/lstchain_create_irf_files.py (hunk 1 above)
Comment (maxnoe) [reply lstchain/tools/lstchain_create_irf_files.py:512]: No, HIERARCH is specifically for longer than 8 character keywords. But it's an extension. We should try to fit in 8 chars.

File: lstchain/io/event_selection.py (hunk 2 above)
Comment (SeiyaNozaki) [reply lstchain/io/event_selection.py:221]: ok, thanks for the comment!

File: lstchain/tools/lstchain_create_irf_files.py (hunk 1 above)
Comment (SeiyaNozaki) [reply lstchain/tools/lstchain_create_irf_files.py]: thanks, I forgot it... I changed it to `AL_CONT`

**Review by chaimain — DISMISSED** [review]

**Review by chaimain — APPROVED** [review]

**Pull request merged**